dataline-workers/src/main/java/io/dataline/workers/protocols/singer/SingerSyncWorker.java

```diff
@@ -68,16 +71,27 @@ public DefaultSyncWorker(TapFactory<SingerMessage> singerTapFactory,
 
     final SingerMessageTracker singerMessageTracker = new SingerMessageTracker();
 
-    try (Stream<SingerMessage> tap = singerTapFactory.create(tapConfig, jobRoot);
-        CloseableConsumer<SingerMessage> consumer = singerTargetFactory.create(targetConfig, jobRoot)) {
+    try (singerTarget; singerTap) {
 
-      tap.takeWhile(record -> !cancelled.get()).peek(singerMessageTracker).forEach(consumer);
+      singerTarget.start(targetConfig, jobRoot);
+      singerTap.start(tapConfig, jobRoot);
+
+      while (!cancelled.get() && singerTap.isFinished()) {
```

Comment:
should this be? `      while (!cancelled.get() && !singerTap.isFinished()) {`

Reply:
:) damn :)

Reply:
I renamed it last minute it was a negative before